### DISCUSSION TITLE
Topk vs search_k (annoy) and ef (hnsw) - what's the difference?

### DISCUSSION THREAD

**Discussion by molokanov50** [milvus-io/milvus#5549] [Q&A and General discussion] [answered]
Hi team,
According to Milvus documentation, `topk` means the number of vectors to return as the most similar ones to the query vector. At the same time, `ef` in hnsw as well as `search_k` in annoy also mean some lookup scope. Where are the vectors from `ef` and `search_k` excerpts used subsequently? If they are returned as the final result, there is some confusion between `topk` and `ef` (`search_k`). Could you please give some explanations about the difference?

**Comment by shengjun1985** [comment, marked as answer]
Hnsw is a graph index.
When querying, the points it accesses is dynamically adjusted.
And the size of its result set will be limited to max(`topk`, 'ef').
In other words, when querying, it will try to find max(`topk`, 'ef') nearest neighbors.

Annoy is a tree index.
When querying, you can specify the number of access points by parameter 'search_k'.
